pkg/harvester/models/harvester/persistentvolumeclaim.js

```diff
@@ -99,24 +102,34 @@ export default class HciPv extends HarvesterResource {
 
   get stateDisplay() {
     const ownedBy = this?.metadata?.annotations?.[HCI_ANNOTATIONS.OWNED_BY];
-    const status = this?.status?.phase === 'Bound' ? 'Ready' : 'NotReady';
+    const volumeError = this.relatedPV?.metadata?.annotations?.[HCI_ANNOTATIONS.VOLUME_ERROR];
+    const degradedVolume = this.relatedPV?.metadata?.annotations?.[HCI_ANNOTATIONS.VOLUME_ERROR] === DEGRADED_ERROR;
+    const status = this?.status?.phase === 'Bound' && !volumeError ? 'Ready' : 'NotReady';
 
     const conditions = this?.status?.conditions || [];
 
     if (findBy(conditions, 'type', 'Resizing')?.status === 'True') {
       return 'Resizing';
-    } else if (ownedBy) {
+    } else if (ownedBy && !volumeError) {
       return 'In-use';
+    } else if (degradedVolume) {
+      return 'Degraded';
     } else {
       return status;
     }
   }
 
   get state() {
-    let status = this?.status?.phase === 'Bound' ? 'Ready' : 'NotReady';
+    const volumeError = this.relatedPV?.metadata?.annotations?.[HCI_ANNOTATIONS.VOLUME_ERROR];
+    const degradedVolume = this.relatedPV?.metadata?.annotations?.[HCI_ANNOTATIONS.VOLUME_ERROR] === DEGRADED_ERROR;
+    let status = this?.status?.phase === 'Bound' && !volumeError ? 'Ready' : 'NotReady';
 
     const conditions = this?.status?.conditions || [];
 
+    if (degradedVolume) {
+      status = 'Degraded';
+    }
+
```

Comment:
Is this function still needed?  It should be the same as `stateDisplay`

Reply:
It seems like this property is only applied at https://github.com/harvester/dashboard/blob/4fc2a425ffc96fb13ace8f14b2f3628ce0a70420/pkg/harvester/edit/kubevirt.io.virtualmachine/VirtualMachineVolume/index.vue#L271

Reply:
After discussion, we should retain the state and stateDisplay,  @DaiYuzeng   add a comment in the state function, and analyze after the release.  thanks